src/bindings/python/src/pyopenvino/graph/dict_attribute_visitor.cpp

```diff
@@ -274,6 +274,18 @@ void util::DictAttributeSerializer::on_adapter(const std::string& name, ov::Valu
     if (m_attributes.contains(name)) {
         OPENVINO_THROW("No AttributeVisitor support for accessing attribute named: ", name);
     }
+    
+    if (auto _adapter = dynamic_cast<ov::AttributeAdapter<std::shared_ptr<ov::op::util::Variable>>*>(&adapter)) {
+        m_attributes[name.c_str()] = _adapter->get()->get_info().variable_id;
+    } else if (auto _adapter = dynamic_cast<ov::AttributeAdapter<ov::PartialShape>*>(&adapter)) {
+        auto partial_shape = _adapter->get();
+        py::list shape;
```

Comment:
```suggestion
        py::list shape;
```
Is it necessary to use `py::list`, can it be `std::vector<...>` instead?

Reply:
As the test code provided by @akuporos shows, `attributes["shape"]` seems to be `list` type in python, so i implemented that way.

By the way, as `node.partial_shape` will return a `PartialShape` object instead of a `list`, so im wondering if this attribute in attributes_dict should also be a `PartialShape`.

Reply:
> As the test code provided by @akuporos shows, attributes["shape"] seems to be list type in python, so i implemented that way.

[Pybind ](https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html) can simply cast std::vector->list

Reply:
I am not so familiar with `pybind` before, thats why i use this way.

I'll substitute it with `std::vector<int>`, thanks for the advice!

Reply:
@BHbean instead of using simple `int` you should be using the same value the `ov::Dimension` is using. If you use `int` this will result in loosing data if dimension exceeds the range.
Take a look here: https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/core/dimension.hpp#L28

The best practice would be using `ov::Dimension::value_type` directly which can be obtained from this header. (PS this is why few CIs are failing for you)

Reply:
Thanks for the advice! I'll replace `int` with `ov::Dimension::value_type`! (PS actually im really confused about why some CIs failed, thanks for telling me the reason)